hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -589,14 +595,23 @@ private boolean moveContainer(DatanodeDetails source,
       Set<DatanodeDetails> selectedTargets,
       ContainerMoveSelection moveSelection, DatanodeDetails source) {
     countDatanodesInvolvedPerIteration += 2;
+    // don't count a source that has been involved in move earlier
+    if (sourceToTargetMap.containsKey(source) ||
+        selectedTargets.contains(source)) {
+      countDatanodesInvolvedPerIteration -= 1;
+    }
+    // don't count a target that has been involved in move earlier
+    if (selectedTargets.contains(moveSelection.getTargetNode())) {
+      countDatanodesInvolvedPerIteration -= 1;
+    }
```

Comment:
```suggestion
    // don't count a source that has been involved in move earlier
    if (sourceToTargetMap.containsKey(source) ||
        selectedTargets.contains(source)) {
      countDatanodesInvolvedPerIteration++;
    }
    // don't count a target that has been involved in move earlier
    if (selectedTargets.contains(moveSelection.getTargetNode())) {
      countDatanodesInvolvedPerIteration++;
    }
```
NIT

Reply:
The suggestion looks wrong semantically since we want to exclude re-counting a node that we have already counted earlier. For example in the suggested code, if both the if conditions are satisfied, the result will equal 2. But the correct result should be 0.

Reply:
sorry, i made a mistake here
```suggestion
    // don't count a source that has been involved in move earlier
    if (!sourceToTargetMap.containsKey(source) &&
        !selectedTargets.contains(source)) {
      countDatanodesInvolvedPerIteration++;
    }
    // don't count a target that has been involved in move earlier
    if (&selectedTargets.contains(moveSelection.getTargetNode())) {
      countDatanodesInvolvedPerIteration++;
    }
```

Reply:
Done!